### PR TITLE
MRD-2608 Adding optional text for ppud index offence

### PR DESCRIPTION
**Pull request by MancRickster**
Allow optional text comment when entering ppud index offence

### CODE REVIEW

File: server/controllers/recommendation/matchIndexOffenceController.test.ts [hunk 1]
```diff
@@ -144,6 +146,7 @@ describe('post', () => {
         bookRecallToPpud: {
           policeForce: 'Kent',
           indexOffence: 'some offence',
+          indexOffenceComment: null,
```
Comment (ppmg) [inline server/controllers/recommendation/matchIndexOffenceController.test.ts:149]: Can you add tests for the empty string and whitespace cases too, please, to make sure it's setting `indexOffenceComment` to `null` when expected?

**Review by ppmg — CHANGES_REQUESTED** [review]

**Review by ppmg — APPROVED** [review]
LGTM
